sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -241,4 +242,33 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
     val alteredTable = externalCatalog.getTable("db1", tableName)
     assert(DataTypeUtils.sameType(alteredTable.schema, newSchema))
   }
+
+  test("SPARK-50137: Avoid fallback to Hive-incompatible ways on thrift exception") {
+    val hadoopConf = new Configuration()
+    // Use an unavailable uri to mock client connection timeout.
+    hadoopConf.set("hive.metastore.uris", "thrift://1.1.1.1:1111")
```

Comment:
Oh, can we avoid to use `1.1.1.1` here, @wecharyu and @sunchao ? This looks like a DDoS attach to `1.1.1.1`.
```
$ ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1): 56 data bytes
64 bytes from 1.1.1.1: icmp_seq=0 ttl=59 time=3.495 ms
64 bytes from 1.1.1.1: icmp_seq=1 ttl=59 time=4.108 ms
```

Reply:
1.1.1.1 is the valid DNS server of `Cloudflare` like Google's 8.8.8.8.

Given that Apache Spark test cases are running massively (not only in Apache Spark repo, but for all PRs and downstream forks), I'd recommend to revert this PR first, @sunchao .

Reply:
thanks for checking this @dongjoon-hyun - yes please revert for now.

Reply:
Got it. I reverted here.
- https://github.com/apache/spark/commit/701e356c534fba2cf30ffcb1ed9513c6f74bf14c

Reply:
Thanks for pointing that out. Change the real ip to a reserved address `192.0.2.1` in #50985